syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.helloworld";
option java_outer_classname = "HelloWorldProto";
option objc_class_prefix = "HLW";

package helloworld;

// The greeting service definition.
service Greeter {
  rpc UploadFile (UploadRequest) returns (UploadResponse) {}
}

// Define the message for upload request
message UploadRequest {
  repeated string filename = 1;
}


// Define the message for upload response
message UploadResponse {
  bool success = 1;
  string message = 2; // Optional message for success or error details
  string debug_log = 3; // New field for debug log
}
